### DISCUSSION TITLE
Forcepoint extension blocks chrome POST requests in test mode.

### DISCUSSION THREAD

**Discussion by maromg** [cypress-io/cypress#9179] [Questions and Help]
I have an issue with the combination of Forcepoint extension that is installed on our workplace env.

It seems that the extension works  when chrome is opened normally.
When chrome is opened in test mode, all post requests ( regardless of whether its a test or a normal site ) are getting blocked by the Forcepoint endpoint (blocked:other).

What should be whitelisted inside of the Forcepoint configuration in order for this not to happen?

**Comment by vladpereverzyev** [comment]
This sounds more like an interaction between Cypress's Chrome test environment and the Forcepoint extension than a Cypress-specific issue with POST requests.

When Cypress launches Chrome, it does not use exactly the same browser environment as a normal interactive Chrome session. It can use a separate profile, different command-line flags, and an automation/test context. An enterprise security extension may therefore classify or intercept those requests differently.

I would troubleshoot it in this order:

1. **Check the exact request being blocked**

   Open Chrome DevTools while running Cypress and inspect one of the failed POST requests:

   * Request URL
   * Request method
   * Response/status
   * Request headers
   * Whether the request is blocked before it reaches the server

   If Forcepoint reports `blocked:other`, the Forcepoint logs would be particularly useful because that classification should tell you what policy triggered the block.

2. **Compare normal Chrome vs Cypress Chrome**

   Make the same POST request from:

   * normal Chrome
   * Chrome launched by Cypress

   If the URL, headers and payload are identical but only the Cypress-launched browser is blocked, that strongly points to the browser/extension context rather than the application itself.

3. **Check the Forcepoint policy/logs**

   I would not whitelist an entire domain immediately.

   First determine whether Forcepoint is blocking based on:

   * destination domain
   * URL/path
   * HTTP method
   * browser/extension context
   * user agent
   * certificate/TLS inspection
   * Chrome automation flags

   Once the exact policy rule is identified, the appropriate exception can be added by whoever administers Forcepoint.

4. **Check whether the Forcepoint extension is actually required during Cypress tests**

   If the extension is only required for normal corporate browsing and isn't needed for the application's automated tests, the cleanest solution may be to run Cypress in an environment where the extension is not installed/enforced.

   However, I would only do this if it complies with the organization's security policy. I would not recommend trying to bypass the Forcepoint controls from Cypress itself.

5. **Test with Cypress's Electron browser**

   As a diagnostic step, try running the same test using Cypress's Electron browser instead of Chrome.

   If the POST requests work in Electron but fail only in Chrome, that provides strong evidence that the issue is specifically related to Chrome + the Forcepoint extension.

6. **Check Cypress/Chrome compatibility**

   This discussion is from November 2020, so I would also verify the exact Cypress and Chrome versions. Browser automation behavior and extension APIs have changed significantly since then.

My recommendation would therefore be to **identify the exact Forcepoint rule that produces `blocked:other` rather than trying to whitelist a generic Cypress URL**.

If the same POST works in normal Chrome but is blocked only when Chrome is launched by Cypress, the Forcepoint administrator should compare the two browser contexts and determine which policy condition changes.

I would also avoid disabling Forcepoint or attempting to circumvent the policy in the Cypress configuration. The proper fix is an explicit, narrowly scoped exception in the Forcepoint policy if the organization's security team approves it.
